### ISSUE TITLE
Credential type 'telldus-gateway' is not registered

### ISSUE THREAD

**Issue by Nauticno** [perbrage/node-red-contrib-telldus-zwave-v2-local#1]
I have trouble with registration. I have my Node-RED installed in a container in docker. I am able to get key when I am in the container. So there should not be a networking issue.

Trying by adding my ip for TellStick ZNet Lite v2. It will not work. What am I doing wrong?

![image](https://user-images.githubusercontent.com/22323906/38200802-08e15230-3696-11e8-8e71-4c382613e2a5.png)

**Comment by perbrage** [comment]
From your picture, did you click the button next to your filled in IP?

Also, make sure you have the latest version of this npm, there was perviously some bugs regarding this wizard.

**Comment by Nauticno** [comment]
Yes, I did click the button next to IP. Nothing happens. I have now started Nodered from scratch and installed
![image](https://user-images.githubusercontent.com/22323906/38204893-eef45002-36a4-11e8-93ad-31b2d6ee67b4.png)

And then I run:
node-red@70a54662ea16:~$ npm install node-red-contrib-telldus-zwave-v2-local
node-red-docker@1.0.0 /usr/src/node-red
`-- node-red-contrib-telldus-zwave-v2-local@1.3.3
  +-- promise@8.0.1
  | `-- asap@2.0.6
  `-- telldus-local@0.0.3

**Comment by Nauticno** [comment]
I'm using Google Chrome. Could it be something with the browser?

**Comment by Nauticno** [comment]
I also have another problem. The Telldus palette disapear. Is this a problem with Telldus palette or is this a NodeRED trouble?

**Comment by perbrage** [comment]
1.3.3 is the latest version and I have verified the nodes UI/Wizard works with Google Chrome. 

I am not sure I understand what you mean with the Telldus palette disappear? But if you have installed the nodes properly (which I assume you have, since you are posting pictures from them) they should not disappear. 

Have you tried using the palette to install this package, instead of using npm? Also, if you are on a later NodeRed and have turned on projects (done in settings file), you might need to install them to the project as well as to your NodeRed installation.

**Comment by Nauticno** [comment]
Hmm. I'm using Docker and Node red is in a container. Can that be a problem?
I have recreatede my Node red container and just added the palett. What is suppose to happen when I push the button? Why is nothing happening? I user NodeRED version 0.18.4, no project (as I know of)

**Comment by perbrage** [comment]
When you push the button, you should either get a message as status right under the textbox telling you that it can not find the telldus gateway, or if the telldus gateway was found, part 2 of the wizard will be lit up with an authorize button. You should get either of these when clicking the button. 

I am running 0.18.4 as well!

Are you sure you can reach the telldus gateway from within the docker container? Can you ping it? Can you use other network-related nodes?

**Comment by perbrage** [comment]
You could also try to use the link in the tip box at the bottom and manually curl a access token and fill that in with the IP in the two bottom textboxes to see if that works.

**Comment by Nauticno** [comment]
I am sure that docker container can access telldus. When I use curl I get tokens. I have put them in the textboxes with no luck. Is there a logfile I can view?

**Comment by Nauticno** [comment]
I'm getting :
msg : string[51]
"Credential type 'telldus-gateway' is not registered"

**Comment by perbrage** [comment]
As far as I know, that error message is coming from NodeRED, not the telldus contrib. It basically says NodeRED can not find all parts of the contrib. "telldus-gateway" is the configuration node, which would be created by using and completing the dialog you originally posted above.

My best guess is that something has gone wrong in the installation, or while updating the lib. Either it failed installing properly, or you have mixed versions in your flows. I noticed that references to old nodes sometimes gets stuck in flows, and I have to manually edit the saved files to completely remove old versions. Seems like a bug in NodeRED.

I haven't really added logging to these nodes yet, but that could be something for the next version.

**Comment by perbrage** [comment]
Depending on the your configuration, but you will find the flow files stored in ~/.node-red/ or ~/.node-red/projects/<name of project>/

They are named something like..  flows_<computername>.json

Uninstall the telldus contrib, edit these files and remove anything left regarding the telldus contrib. Restart and try to reinstall the contrib.

**Comment by perbrage** [comment]
Closing this soon as I haven't heard anything in a long time
